server/routes/bookVisit/selectVisitorsController.ts

```diff
@@ -35,14 +41,15 @@ export default class SelectVisitorsController {
   public submit(): RequestHandler {
     return async (req, res) => {
       const errors = validationResult(req)
+
       if (!errors.isEmpty()) {
         req.flash('errors', errors.array())
-        req.flash('formValues', req.body)
+        req.flash('formValues', matchedData(req, { onlyValidData: false }))
         return res.redirect('/book-visit/select-visitors')
       }
 
       const { bookingJourney } = req.session
-      const { visitorDisplayIds }: { visitorDisplayIds: number[] } = req.body
+      const { visitorDisplayIds } = matchedData<{ visitorDisplayIds: number[] }>(req)
```

Comment:
Using `matchedData<T>()` to get the valid data - taking advantage of generic return type to define what the validated data will be like.